packages/admin-panel/src/qrCode/QrCode.js

```diff
@@ -0,0 +1,98 @@
+/*
+ * Tupaia
+ * Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
+ */
+
+import React, { useEffect, useRef } from 'react';
+import styled from 'styled-components';
+import PropTypes from 'prop-types';
+import { toDataURL } from 'qrcode';
+import { Button } from '@tupaia/ui-components';
+
+const Container = styled.div`
+  text-align: center;
+`;
+
+const ButtonContainer = styled.div`
+  margin-top: 30px;
+`;
+
+const CODE_SIZE = 320; // width/height at browser display size
+const CANVAS_SIZE = 395; // width/height at browser display size
+const DOWNLOAD_IMAGE_SCALE = 2; // render the canvas at double size to get a larger and clearer image download
+
+const StyledCanvas = styled.canvas`
+  width: ${CANVAS_SIZE}px;
+  height: ${CANVAS_SIZE}px;
+  outline: 1px solid #dedede;
+`;
+
+function downloadImage(dataUri, filename) {
+  const a = document.createElement('a');
+  a.href = dataUri;
+  a.download = filename;
+  a.click();
+}
+
+export const QrCode = ({ qrCodeContents, humanReadableId }) => {
+  // Draw into a canvas so that we can download it as an image
+  const canvasRef = useRef(null);
+
+  const handleDownload = () => {
+    const canvas = canvasRef.current;
+    const dataUri = canvas.toDataURL('image/jpeg', 1.0);
+    downloadImage(dataUri, `${humanReadableId}.jpeg`);
```

Comment:
```suggestion
    downloadJs(dataUri, `${humanReadableId}.jpeg`);
```
Looks like we use [downloadJs](https://www.npmjs.com/package/downloadjs) for image downloads in other places

Reply:
Nice catch, updated